Source/Core/Core/ConfigManager.cpp

```diff
@@ -889,154 +890,93 @@ bool SConfig::SetRegion(DiscIO::Region region, std::string* directory_name)
-      // the emulator config.
-      SetRegion(bWii ? DiscIO::Region::PAL : DiscIO::Region::NTSC_U, &set_region_dir);
-      m_BootType = BOOT_ELF;
-    }
-    else if (!strcasecmp(Extension.c_str(), ".dol"))
-    {
-      CDolLoader dolfile(m_strFilename);
-      bWii = dolfile.IsWii();
-      // TODO: See the ELF code above.
-      SetRegion(bWii ? DiscIO::Region::PAL : DiscIO::Region::NTSC_U, &set_region_dir);
-      m_BootType = BOOT_DOL;
-    }
-    else if (!strcasecmp(Extension.c_str(), ".dff"))
-    {
-      bWii = true;
-      SetRegion(DiscIO::Region::NTSC_U, &set_region_dir);
-      m_BootType = BOOT_DFF;
+    config->SetRunningGameMetadata(tmd);
+    config->bWii = true;
+    *region = tmd.GetRegion();
+    return true;
+  }
 
-      std::unique_ptr<FifoDataFile> ddfFile(FifoDataFile::Load(m_strFilename, true));
+  bool operator()(const BootParameters::IPL& ipl) const
+  {
+    config->bWii = false;
+    *region = ipl.region;
+    return true;
+  }
 
-      if (ddfFile)
-      {
-        bWii = ddfFile->GetIsWii();
-      }
-    }
-    else if (DiscIO::CNANDContentManager::Access().GetNANDLoader(m_strFilename).IsValid())
-    {
-      const DiscIO::CNANDContentLoader& content_loader =
-          DiscIO::CNANDContentManager::Access().GetNANDLoader(m_strFilename);
-      const IOS::ES::TMDReader& tmd = content_loader.GetTMD();
+  bool operator()(const BootParameters::DFF& dff) const
+  {
+    std::unique_ptr<FifoDataFile> dff_file(FifoDataFile::Load(dff.dff_path, true));
+    if (!dff_file)
+      return false;
 
-      if (!IOS::ES::IsChannel(tmd.GetTitleId()))
-      {
-        PanicAlertT("This WAD is not bootable.");
-        return false;
-      }
+    config->bWii = dff_file->GetIsWii();
+    *region = DiscIO::Region::NTSC_U;
+    return true;
+  }
 
-      SetRegion(tmd.GetRegion(), &set_region_dir);
-      SetRunningGameMetadata(tmd);
+private:
+  SConfig* config;
+  DiscIO::Region* region;
+};
 
-      bWii = true;
-      m_BootType = BOOT_WII_NAND;
-    }
-    else
-    {
-      PanicAlertT("Could not recognize ISO file %s", m_strFilename.c_str());
-      return false;
-    }
-  }
-  break;
-
-  case BOOT_BS2_USA:
-    SetRegion(DiscIO::Region::NTSC_U, &set_region_dir);
-    m_strFilename.clear();
-    m_BootType = SConfig::BOOT_BS2;
-    break;
-
-  case BOOT_BS2_JAP:
-    SetRegion(DiscIO::Region::NTSC_J, &set_region_dir);
-    m_strFilename.clear();
-    m_BootType = SConfig::BOOT_BS2;
-    break;
-
-  case BOOT_BS2_EUR:
-    SetRegion(DiscIO::Region::PAL, &set_region_dir);
-    m_strFilename.clear();
-    m_BootType = SConfig::BOOT_BS2;
-    break;
+bool SConfig::SetPathsAndGameMetadata(const BootParameters& boot)
+{
+  m_is_mios = false;
+  m_disc_booted_from_game_list = false;
+  DiscIO::Region region;
+  if (!std::visit(SetGameMetadata(this, &region), boot.parameters))
+    return false;
+
+  // Set up region
+  std::string set_region_dir;
+  if (!SetRegion(region, &set_region_dir) && !bWii &&
```

Comment:
We care about the region of Wii titles more than the console does, since we set the region of the hardware based on the region of what we're booting. `SetupWiiMemory` will even crash if the region is invalid (though that's not a problem in this case since we always set it to PAL here if it's invalid).

It's possible to trigger the same problem with GC software - the Datel discs don't respect the game ID conventions, for instance.

Maybe it would make sense to never show this warning, but I don't see why it makes sense to show it for GC but not Wii. Either way, if you're going to change the behavior of the warning, could you do it in a separate commit or PR?